tests/model_registry/model_catalog/upgrade/test_catalog_upgrade.py

```diff
@@ -0,0 +1,429 @@
+        return ConfigMap(name="model-catalog-default-sources", client=admin_client, namespace=model_registry_namespace)
+
+    @pytest.mark.post_upgrade
+    def test_verify_model_catalog_sources_configmap_exists_post_upgrade(
+        self,
+        admin_client: DynamicClient,
+        model_registry_namespace: str,
+    ):
+        """Verify model-catalog-sources ConfigMap exists after upgrade"""
+        LOGGER.info("Starting post-upgrade configmap migration verification")
+
+        model_catalog_sources_cm = self._get_model_catalog_sources_cm(
+            admin_client=admin_client, model_registry_namespace=model_registry_namespace
+        )
+        assert model_catalog_sources_cm.exists, "model-catalog-sources ConfigMap should exist after upgrade"
+
+    @pytest.mark.post_upgrade
+    def test_verify_model_catalog_sources_structure_post_upgrade(
+        self,
+        admin_client: DynamicClient,
+        model_registry_namespace: str,
+    ):
+        """Verify model-catalog-sources ConfigMap has correct structure"""
+        model_catalog_sources_cm = self._get_model_catalog_sources_cm(
+            admin_client=admin_client, model_registry_namespace=model_registry_namespace
+        )
+        sources_data = yaml.safe_load(model_catalog_sources_cm.instance.data["sources.yaml"])
+
+        assert "catalogs" in sources_data, "sources.yaml should contain 'catalogs' key"
+
+    @pytest.mark.post_upgrade
+    def test_verify_custom_catalog_properties_post_upgrade(
+        self,
+        admin_client: DynamicClient,
+        model_registry_namespace: str,
+    ):
+        """Verify custom catalog properties are correct (if custom catalog exists)"""
+        model_catalog_sources_cm = self._get_model_catalog_sources_cm(
+            admin_client=admin_client, model_registry_namespace=model_registry_namespace
+        )
+        sources_data = yaml.safe_load(model_catalog_sources_cm.instance.data["sources.yaml"])
+        catalogs = sources_data["catalogs"]
+
+        # If our test catalog exists, verify it's properly structured
+        test_catalog_found = False
+        for catalog in catalogs:
+            if catalog.get("id") == TEST_DATA["catalog_id"]:
+                test_catalog_found = True
+                # Single assertion combining both checks
+                assert catalog.get("type") == "yaml" and "name" in catalog, (
+                    f"Custom catalog should be yaml type with name. "
+                    f"Got type: {catalog.get('type')}, has name: {'name' in catalog}"
+                )
+                break
+
+        # Only assert if we found the test catalog (it might not exist if pre-upgrade didn't run)
+        if not test_catalog_found and catalogs:
+            LOGGER.info("Custom test catalog not found, but other catalogs exist - this is acceptable")
+
+    @pytest.mark.post_upgrade
+    def test_verify_model_catalog_default_sources_configmap_exists_post_upgrade(
```

Comment:
This is part of test_config_map_exists(). If you use the same custom catalog for the upgrade flow, then you won't need to repeat the same validation.